public/app/features/chat/Chat.tsx

```diff
@@ -0,0 +1,293 @@
+import React, { FunctionComponent, PureComponent, useState } from 'react';
+import { TextArea, IconButton, WithContextMenu, MenuItem, MenuGroup } from '@grafana/ui';
+import { getGrafanaLiveSrv, getBackendSrv } from '@grafana/runtime';
+import { isLiveChannelMessageEvent, LiveChannelScope } from '@grafana/data';
+import { Unsubscribable } from 'rxjs';
+import { marked } from 'marked';
+import { sanitize } from '@grafana/data/src/text/sanitize';
+
+export interface ChatProps {
+  contentTypeId: number;
+  objectId: string;
+  actions?: ChatMessageAction[];
+}
+
+export interface User {
+  id: number;
+  name: string;
+  login: string;
+  email: string;
+  avatarUrl: string;
+}
+
+export interface Message {
+  id: number;
+  content: string;
+  created: number;
+  userId: number;
+  user: User;
+}
+
+export interface MessagePacket {
+  event: string;
+  messageCreated: Message;
+}
+
+export interface ChatState {
+  messages: Message[];
+  value: string;
+}
+
+function renderChatMarkdown(str?: string): string {
+  const html = marked(str || '', {
+    pedantic: false,
+    gfm: true,
+    smartLists: true,
+    smartypants: false,
+    xhtml: false,
+    breaks: true,
+  });
+  return sanitize(html);
+}
+
+export class Chat extends PureComponent<ChatProps, ChatState> {
+  subscription?: Unsubscribable;
+  chatInput?: any;
+  chatContainer?: any;
+
+  state: ChatState = {
+    messages: [],
+    value: '',
+  };
+
+  constructor(props: ChatProps) {
+    super(props);
+  }
+
+  async componentDidMount() {
+    const resp = await getBackendSrv().post('/api/chats/get-messages/', {
+      objectId: this.props.objectId,
+      contentTypeId: this.props.contentTypeId,
+    });
+    this.setState({
+      messages: resp.chatMessages,
+    });
+    this.updateSubscription();
+    this.scrollToBottom();
+    this.focusInput();
+  }
+
+  handleChange = (e: any) => {
+    this.setState({ value: e.target.value });
+  };
+
+  sendMessage = async () => {
+    if (!this.state.value) {
+      return;
+    }
+    await getBackendSrv().post('/api/chats/send-message/', {
+      objectId: this.props.objectId,
+      contentTypeId: this.props.contentTypeId,
+      content: this.state.value,
+    });
+    this.setState({
+      value: '',
+    });
+  };
+
+  getLiveAddr = () => {
+    return {
+      scope: LiveChannelScope.Grafana,
+      namespace: 'chat', // holds on to the last value
+      path: `${this.props.contentTypeId}/${this.props.objectId}`,
+    };
+  };
+
+  getLiveChannel = () => {
+    const live = getGrafanaLiveSrv();
+    if (!live) {
+      console.error('Grafana live not running, enable "live" feature toggle');
```

Comment:
In general we will want to go through and add some error handling UX on the event the service is down / user is not connected for whatever reason and whatever other edge cases we can think of.

Given this is early not necessarily a blocker for this PR, but would be good to consider for when we polish this to go into beta / beyond